File: Week 01/id_003/mergeTwoLists.js
var mergeTwoLists = function(l1, l2) {
    var prehead = {
        val:null,
        next:null
    };
    var pre = prehead;
    while (l1 != null && l2 != null) {
        if ( l1.val <= l2.val ) {
            pre.next = l1;
            l1 = l1.next;
        } else {
            pre.next = l2;
            l2 = l2.next;
        }
        pre = pre.next;
        
    }
    pre.next = l1 == null ? l2 : l1;
    return prehead.next;
};

File: Week 01/id_328/MergeTwoLists.java
public class MergeTwoLists {
     public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode newHead = null;
        ListNode tmp = new ListNode(-1);
        boolean firstSetNewHead = true;
        while (l1 != null || l2 != null) {
            if(l1 == null) {
                tmp.next = l2;
                l2 = l2.next;
            }else if(l2 == null) {
                tmp.next = l1;
                l1 = l1.next;
            }else {
                if (l1.val > l2.val) {
                    tmp.next = l2;
                    l2 = l2.next;
                } else {
                    tmp.next = l1;
                    l1 = l1.next;
                }
               
            }
            if(firstSetNewHead) {
                newHead = tmp;
                firstSetNewHead = false;
            }
            tmp = tmp.next;
        }
        if(newHead == null) {
            return null;
        }
        return newHead.next;
    }
}